plugins/linux/ifplugin/descriptor/watcher.go

```diff
@@ -204,32 +198,26 @@ func (w *InterfaceWatcher) processLinkNotification(linkUpdate netlink.LinkUpdate
 		// do not notify until interface is truly finished
 		w.pendingIntfs[ifName] = true
 		w.wg.Add(1)
-		go w.delayNotification(ifName)
+		go w.applyDelayedNotification(ifName)
 		return
 	}
 
 	// notification about removed interface is propagated immediately
 	w.notifyScheduler(ifName, false)
 }
 
-// delayNotification delays notification about enabled interface - typically
-// interface is created in multiple stages and we do not want to notify scheduler
-// about intermediate states.
-func (w *InterfaceWatcher) delayNotification(ifName string) {
```

Comment:
Don't remember the exact scenario(s), but the reason for this was that newly created interface would send multiple UP/DOWN notifications, as if it was being created in multiple steps (causing dependent stuff to be created/deleted multiple times). Actually I think it is TAP that in VPP is being created through multiple syscalls. Anyway, let's get rid of the timeout and see if any issues appear (if yes, we can do this for example for VPP TAP, but not for VETH).
Btw. I don't think pendingIntfs map and running applyDelayedNotification as go routine is needed anymore. Also "applyDelayedNotification" is probably not a good name anymore.

Reply:
>  Don't remember the exact scenario(s), but the reason for this was that newly created interface would send multiple UP/DOWN notifications, as if it was being created in multiple steps (causing dependent stuff to be created/deleted multiple times).

For simple case of VETHs+AFPACKET I see several notifications, most likely because of various things configured on the interfaces (alias, namespace, peer, ...) and these dont see to cause issues for me. Running more e2e tests to confirm this though.

> Actually I think it is TAP that in VPP is being created through multiple syscalls.

Will run e2e tests for TAPs as well.

> Anyway, let's get rid of the timeout and see if any issues appear (if yes, we can do this for example for VPP TAP, but not for VETH).

I think we will need to find a way to avoid sleeps/delays totally and somehow figure out which inter-states to ignore, because these sleeps make configuration of interfaces veeeery slow (see #1739)

> Btw. I don't think pendingIntfs map and running applyDelayedNotification as go routine is needed anymore. Also "applyDelayedNotification" is probably not a good name anymore.

Will fix that before merge, just wanted travis to run tests on this.